metricshub-programmable-configuration-extension/src/test/java/org/metricshub/programmable/configuration/ProgrammableConfigurationProviderTest.java

```diff
@@ -132,4 +136,51 @@ void testLoadUsingFileTool() {
 			"Should load correct configuration fragment using file tool"
 		);
 	}
+
+	@Test
+	void testLoadUsingSqlTool() throws SQLException {
+		Connection connection = null;
+		try {
+			// Setup in-memory H2 database and create table
+			String url = "jdbc:h2:mem:testdb1";
+			connection = DriverManager.getConnection(url, "sa", "pwd1");
+
+			try (Statement statement = connection.createStatement()) {
+				String createTableSQL =
+					"""
+					    CREATE TABLE IF NOT EXISTS users (
+					      username VARCHAR(64),
+					      password VARCHAR(64),
+					      hostname VARCHAR(128),
+					      ostype   VARCHAR(32)
+					    )
+					""";
+				statement.execute(createTableSQL);
+
+				// Insert data
+				String insertDataSQL =
+					"""
+					    INSERT INTO users (username, password, hostname, ostype) VALUES
+					      ('adminA','passA','host-a','STORAGE'),
+					      ('adminB','passB','host-b','LINUX'),
+					      ('userC','passC','host-c','WINDOWS')
+					""";
+				statement.execute(insertDataSQL);
+			}
+
+			var provider = new ProgrammableConfigurationProvider();
+			var nodes = provider.load(Paths.get("src/test/resources/sql"));
+
+			assertEquals(
+				"""
+				[{"resources":{"host-a":{"attributes":{"host.name":"host-a","host.type":"STORAGE"},"protocols":{"http":{"hostname":"host-a","https":true,"port":443,"username":"adminA","password":"passA","timeout":300}},"connectors":["+DellEMCPowerStoreREST"],"sequential":true}}}]""",
+				nodes.toString(),
+				"Should load one configuration fragment with sql tool"
+			);
+		} finally {
+			if (connection != null) {

```

Comment:
Connection is AutoClosable, use the try-with-resources statement so that you can remove this finally block.